player.go

```diff
@@ -36,7 +36,6 @@ func newPlayer(context *Context) *Player {
 		r:       r,
 		w:       w,
 	}
-	runtime.SetFinalizer(p, (*Player).Close)
```

Comment:
Why did you remove this? Without this, players can leak.

Reply:
Sure, but with the `SetFinalizer` when we need to close the Player manually before closing the Context or we get a `mux: already closed` error.

Reply:
And in my mind, it's the user's job to make sure the program doesn't leak, not the lib itself.

Reply:
> And in my mind, it's the user's job to make sure the program doesn't leak, not the lib itself.

Ensuring to call Close is a conventional way in Go (e.g., os.File).

Reply:
> Sure, but we need to close the Player manually before closing the Context or we get a mux: already closed error.

If so, what we should do would be close all the players internally when closing the context.

Reply:
> close all the players internally when closing the context

That's a better way IMO

Reply:
OK, so please update the PR to do so :-)